### PR TITLE
Merge SARIF reports

### PR DESCRIPTION
**Pull request by chao2zhang**
This PR addresses the last item in #3360

## Notes
1. Although Gradle repackages Jackson dependency internally, its version seems outdated and I encountered two parsing issues. Therefore, I am adding the latest Jackson dependency explicitly.

2. `-Dorg.gradle.jvmargs=-Xmx2g -XX:MaxMetaspaceSize=512m` becomes necessary. It turns out that `DetektSarifReportMergeTest.kt` runs out of Metaspace as OutOfMemoryError with Java 11 + Ubuntu/Windows. Hence I am bumping the Xmx and MaxMetaspaceSize slightly for test to pass.

### CODE REVIEW

File: detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/report/SarifReportMergeTask.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    abstract val input: ConfigurableFileCollection
+
+    @get:OutputFile
+    abstract val output: RegularFileProperty
+
+    @TaskAction
+    fun merge() {
+        logger.info(input.files.joinToString(separator = "\n") { it.absolutePath })
+        SarifReportMerger.merge(input.files, output.get().asFile)
+        logger.lifecycle("Merged SARIF output to ${output.get().asFile.absolutePath}")
```
Comment (cortinico) [inline detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/report/SarifReportMergeTask.kt:27]: This basically `XmlReportMergeTask` modulo the `SarifReportMerger`.

Could we refactor this a bit? Say define a `ReportMerger` Interface + a `ReportMergeTask` that takes an `enum` as input property?

We could probably have other mergers (HTML?) so it's better we clean it up a bit before we start exposing this 🤔 ?

File: docs/pages/reporting.md [hunk 2]
```diff
@@ -77,11 +77,13 @@ The machine-readable report formats support report merging.
 Detekt Gradle plugin is not opinionated in how merging is set up and respects each project's build logic, especially 
 the merging makes most sense in a multi-module project. In this spirit, only Gradle tasks are provided.
 
-At the moment, only merging XML is supported. You can refer to the sample script below and 
-run `./gradlew detekt xmlReportMerge --continue` to execute detekt tasks and merge their reports.
+At the moment, merging XML and SARIF are supported.
```
Comment (cortinico) [inline docs/pages/reporting.md:80]:
```suggestion
At the moment, only merging XML and SARIF are supported. 
```

**Review by cortinico — APPROVED** [review]
LGTM 👍 I've left a minor comment for a refactoring. But we can also skip it if we wish

File: detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/report/SarifReportMergeTask.kt (hunk 1 above)
Comment (chao2zhang) [reply detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/report/SarifReportMergeTask.kt:27]: Good idea. We could potentially avoid duplication in docs as well.

**Pull request merged**